dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/sampling/Sampler.kt

```diff
@@ -0,0 +1,6 @@
+package com.datadog.android.core.internal.sampling
+
+internal interface Sampler {
+
+    fun sample(): Boolean
```

Comment:
I think eventually we'll need to have a sampler based on the data to be sampled, eg : 

```kotlin
internal interface Sampler<T: Any> {
    fun sample(data: T): Boolean
}
```

But it might be overkill for now, we can change that later since it's internal

Reply:
Also maybe add a javadoc on the method to say what the return value means ( `true` -> the data is kept).

Reply:
I had this in the fist version...but then ...I didn't want to introduce this now....also because this will make me build a Log object just to pass it to the sampler. Without that I can just check the sampler and if returns true I create the Log object (less memory and more optimised). We can easily introduce this later.